quality_check/08-gene_lengths.Rmd

```diff
@@ -0,0 +1,293 @@
+       subtitle = "All genes", title = "Gene length and observations")
+```
+
+```{r}
+ggsave(filename = file.path(plot_dir, 
+                            "gene_length_vs_zero_count_all_genes.png"),
+       plot = last_plot())
+```
+
+#### Compendium genes only
+
+```{r}
+# subset just to genes that are in the compendium
+zeroes_df %>%
+  dplyr::filter(Gene %in% compendium_genes, 
+                gene_length <= 20000) %>%
+  ggplot(aes(x = gene_length, y = Zero_proportion)) +
+  geom_point(alpha = 0.2) +
+  geom_smooth() +
+  theme_bw() +
+  labs(x = "gene length", y = "proportion of samples with zero count",
+       subtitle = "Test compendium genes only", 
+       title = "Gene length and observations")
+```
+
+```{r}
+ggsave(filename = file.path(plot_dir, 
+                            "gene_length_vs_zero_count_compendium_genes.png"),
+       plot = last_plot())
+```
+
+#### Takeaway
+
+Shorter genes do tend to have a larger proportion of samples where they have
+zero values. 
+It stands to reason that if we do not _observe_ shorter genes, we can not
+correct for length-bias effectively. 
+(Recall that `lengthScaledTPM` are calculated using the average tx length
+observed across samples.)
+
+## Removing "short" genes from the test species compendium
+
+Does removing shorter genes from the species compendium remove evidence of a
+technology bias?
+
+```{r}
+exprs_mat <- compendium_exprs_df %>%
+  as.data.frame() %>%
+  tibble::column_to_rownames("X1") %>%
+  as.matrix()
+```
+
+```{r}
+# derive the technology labels from the sample names
+sample_names <- colnames(exprs_mat)
+compendium_technology_labels <- rep("MICROARRAY", length(sample_names))
+compendium_technology_labels[grep("lengthScaledTPM", sample_names)] <- "RNASEQ"
+
+# get the technology labels as a data.frame for plotting
+technology_df <- as.data.frame(compendium_technology_labels) 
+```
+
+```{r}
+# what genes are short? using 25th percentile
+short_genes <- zeroes_df %>%
+  dplyr::filter(gene_length <= quantile(zeroes_df$gene_length, 0.25)) %>%
+  dplyr::pull(Gene)
+```
+
+#### PCA - all genes
+
+First we'll look at PC1 and PC2 for all genes
+
+```{r}
+pca_results <- prcomp(t(exprs_mat))
+# plotting with autoplot for PCA
+autoplot(pca_results, data = technology_df, alpha = 0.5, 
+         colour = 'compendium_technology_labels', scale = 0) +
+  scale_color_manual(labels = c("MICROARRAY", "RNA-SEQ"), 
+                     values = c("#E69F00", "#56B4E9")) +
+  theme_bw() +
+  ggtitle("Test Compendium by Technology: All genes") +
+  theme(plot.title = element_text(hjust = 0.5, face = "bold")) +
+  guides(colour = guide_legend(title = "technology"))
+```
+
+```{r}
+ggsave(filename = file.path(plot_dir, 
+                            "small_test_compendium_PCA.png"),
+       plot = last_plot())
+```
+
+Pairs plot (PC1-5)
+
+```{r}
+png(file.path(plot_dir, "small_test_compendium_pairs_plot.png"))
+pairs(pca_results$x[, 1:5], col = dplyr::recode(compendium_technology_labels,
```

Comment:
Do you apply this recoding step here and then again in line 271? Does it make sense to just apply it once else where and store it as it's own object so you don't have to do it twice? I might be missing something though.

Reply:
Looks like you are just using this so the legend turns out right? I don't know how `pairs` works, but if it's like `ggplot2` stuff, you might be able to just overwrite the legend to be what you want instead of changing the object. That may end up being messier than just changing the `compendium_technology_labels` though. So maybe never mind.